python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -204,12 +207,55 @@ def get_component_types_in_module(
 T = TypeVar("T")
 
 
+@dataclass
+class MultiComponentsLoadContext:
+    """Context for loading an entire component hierarchy in a code location.
+    Stores resources and a cache to ensure we don't load the same component multiple times.
+    """
+
+    resources: Mapping[str, object]
+
+    def build_defs_from_component_path(self, path: Path) -> Definitions:
```

Comment:
I don't think this should be a `Path` but just a string that represents the module path. Might even want it to be python module syntax "subpackage.module_name" in order for it to work on python files and directories.

Reply:
Took a stab at that here https://github.com/dagster-io/dagster/pull/28197

Reply:
can we just make the change here?

Reply:
Updated here, let me know what you think